### PR TITLE
Implement eth/67 sub protocol (EIP-4938)

### LINKED ISSUE #4596: Implement eth/67 sub protocol (EIP-4938)
eth/67 ([EIP-4938](https://eips.ethereum.org/EIPS/eip-4938), March 2022)
Version 67 removed the GetNodeData and NodeData messages.

GetNodeData and NodeData were introduced in protocol version eth/63 to allow for a sync mode called “fast sync”.

Besu:
- Supports multiple versions of a wire protocol. 
- Does not break older clients, since they can keep using protocol version eth/66.
- For fast sync, eth/66 will be used

### PR DESCRIPTION
**Pull request by Gabriel-Trintinalia**
Signed-off-by: Gabriel Trintinalia <gabriel.trintinalia@gmail.com>

## PR description
eth/67 ([EIP-4938](https://eips.ethereum.org/EIPS/eip-4938), March 2022)
Version 67 removed the GetNodeData and NodeData messages.

Besu:

- Supports multiple versions of a wire protocol.
- Does not break older clients, since they can keep using protocol version eth/66.
- For fast sync, eth/66 will be used

## Fixed Issue(s)
fixes #4596 

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if
    [updates are required](https://wiki.hyperledger.org/display/BESU/Documentation).

## Changelog

- [x] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthProtocolManager.java [hunk 1]
```diff
@@ -210,6 +211,14 @@ private List<Capability> calculateCapabilities(
     capabilities.add(EthProtocol.ETH65);
     capabilities.add(EthProtocol.ETH66);
 
+    // Version 67 removes the GetNodeData and NodeData
+    // Fast sync depends on GetNodeData and NodeData
+    // Do not add eth/67 if fast sync is enabled
+    // see https://eips.ethereum.org/EIPS/eip-4938
+    if (!Objects.equals(SyncMode.FAST, synchronizerConfiguration.getSyncMode())) {
```
Comment (macfarla) [inline ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthProtocolManager.java:218]: line 206 checks sync mode for fullSync but it does it differently - would be nice if these two matched. idk which is better

File: ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/manager/EthProtocolManagerTest.java [hunk 2]
```diff
@@ -1117,4 +1120,40 @@ public void forkIdForChainHeadMayBeNull() {
...
     }
   }
+
+  @Test
+  public void shouldUseRightCapabilityDependingOnSyncMode() {
+    assertHighestCapability(SyncMode.X_SNAP, EthProtocol.ETH67);
+    assertHighestCapability(SyncMode.FULL, EthProtocol.ETH67);
+    assertHighestCapability(SyncMode.X_CHECKPOINT, EthProtocol.ETH67);
+    /* Eth67 does not support fast sync, see EIP-4938 */
+    assertHighestCapability(SyncMode.FAST, EthProtocol.ETH66);
```
Comment (macfarla) [inline ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/manager/EthProtocolManagerTest.java:1130]: nice

**Review by macfarla — COMMENTED** [review]
looks nice and simple! Did you test it out on a node running fast sync?

File: ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthProtocolManager.java (hunk 1 above)
Comment (Gabriel-Trintinalia) [reply ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/manager/EthProtocolManager.java:218]: Yes, created some nodes with fast sync: https://github.com/hyperledger/besu/issues/4596

**Review by macfarla — APPROVED** [review]

**Review by diega — APPROVED** [review]

**Pull request merged**